_posts/2023-10-23-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,257 @@
+It will only manage Kafka topics associated with `KafkaTopic` resources, and won't interfere with topics managed independently within the Kafka cluster.
+It can be deployed standalone, or along with your Kafka cluster using the `Kafka` custom resource.
+
+<figure>
+    <img src="/assets/images/posts/2023-10-23-uto-interactions.png" height=440>
+    <figcaption><small>Fig 1. Unidirectional Topic Operator interactions.</small></figcaption>
+</figure>
+
+Users can create and dynamically change Kafka topic configurations by interacting with the `KafkaTopic` custom resources in real-time.
+A reconciliation loop compares the desired state in `KafkaTopic` resources with the current state of Kafka topics in Kafka.
+Actions are taken to align the actual topic configurations with the custom resource specifications.
+If the Kafka-side configuration is changed out-of-band (e.g. by using the Admin client, or Kafka shell scripts), those changes will eventually be reverted.
+
+The operator requires that a single `KafkaTopic` resource is used to manage a single topic in a single Kafka cluster.
+It detects cases where multiple resources are attempting to manage a Kafka topic using the same `spec.topicName`.
+Only the oldest `KafkaTopic` resource will be reconciled, while the other will fail with a `ResourceConflict` error.
+
+In addition to be able to pause reconciliations, the operator also provides a way to unmanage a Kafka topic using annotations.
+If the `KafkaTopic` is paused and we delete it, then it will be also deleted in Kafka.
+If the `KafkaTopic` is unmanaged and we delete it, then only the resource will be garbage collected by Kubernetes, but the topic will still exists in Kafka.
+
+As with previous implementation, we still do not support decreasing `spec.partitions` and changing `spec.replicas`.
+While the former is not supported by Kafka itself, support for the latter could be added in the future.
+
+### Scalability
+
+The BTO is limited in term of scalability as it t only operates on one topic at a time, and it has to update a persistent store containing topic metadata.
+Instead, the UTO does not store topic metadata and it aims to be scalable in terms of the number of topics that it can operate on.
+
+When running Kafka operations, the UTO makes use of the request batching supported by the Kafka Admin client to get higher throughput for metadata operations.
+All `KafkaTopic` events are queued when received, and then processed in batches by a number of controller threads (currently, only a single thread is supported).
+If more than one event related to a single topic resource are found in the batch building, they are put back in the queue to be processed with the next batch.
+
+The following line graph confirms the BTO scalability issue, while the UTO can scale almost linearly.
+Using the UTO with batch size 500 and linger 1000, we were able to achieve a max reconciliation time of 5.3 seconds with 10k concurrent topic events.
```

Comment:
This does not seem to be in the chart. It is also not clear what `batch size 500 and linger 1000`. Maybe you can move this below the chart and incorporate it into the paragraph about these options?